packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -107,6 +107,34 @@ import {
 } from './utils';
 import { showDefaultIdentityClaimWarning, showOwnerCanReassignWarning } from './utils/warnings';
 
+/**
+ * util to get allowed roles for field
+ * if we have a rule like cognito private we can remove all other related roles from the field since it has top level
+ * access by the provider
+ */
+const getReadRolesForField = (acm: AccessControlMatrix, readRoles: Array<string>, fieldName: string): Array<string> => {
+  const hasCognitoPrivateRole = readRoles.some(r => r === 'userPools:private')
+    && acm.isAllowed('userPools:private', fieldName, 'get')
+    && acm.isAllowed('userPools:private', fieldName, 'list')
+    && acm.isAllowed('userPools:private', fieldName, 'sync')
+    && acm.isAllowed('userPools:private', fieldName, 'search')
+    && acm.isAllowed('userPools:private', fieldName, 'listen');
```

Comment:
I'd rewrite this in a slightly more functional way. You could convert this to a small function like 

```typescript
const hasFullReadAccessForRole = (acm: AccessControlMatrix, readRoles: Array<string>, fieldName: string, role: string): boolean => {
  return readRoles.some(r => r === role) && READ_PERMISSIONS.every(p => acm.isAllowed(role, fieldName, p);
};
```

Reply:
Also, is the readRoles.som check required if we're then verifying the acm against all read access types?

Reply:
Ahh nvm, I just saw you moved this. ignore these comments.